fbpic/fields/spectral_grid.py

```diff
@@ -410,3 +432,48 @@ def filter(self, fieldtype) :
                 spectral_rho *= self.filter_array
             else :
                 raise ValueError('Invalid string for fieldtype: %s'%fieldtype)
+                
+                
+                
+        
+        
+class EnvelopeSpectralGrid(SpectralGrid):
+    
+    def __init__(self, kz_modified, kr, m, kz_true, dz, dr,
+                        use_cuda=False ) :
+        
+        SpectralGrid.__init__(self, kz_modified, kr, m, kz_true, dz, dr,
+                        use_cuda=False )
+        Nr, Nz = self.Nr, self.Nz
+        self.A  = np.zeros( (Nz, Nr), dtype='complex' )
+        self.dtA  = np.zeros( (Nz, Nr), dtype='complex' )
+        
+        
+    def push_envelope_with(self, ps):
+        
+            """
+            Push the A and dtA envelope fields over one timestep, using the psatd coefficients.
+        
+            WARNING: currently only implemented for non-comoving simulations, with only CPU usage
+        
+            Parameters
+            ----------
+            ps : PsatdCoeffs object
+                psatd object corresponding to the same m mode
+            """
+            assert (ps.V is None or ps.V == 0)
+            assert( abs(self.m) == ps.m )
+        
+            numba_push_envelope_standard(self.A, self.dtA, ps.w2_square, ps.invw_tot, ps.S_env,
+                ps.C_env, ps.sinc_env, ps.A_coef, self.Nz, self.Nr)
+
+
```

Comment:
For now, I think that the `filter` function is not needed, and so it could be removed.